packages/tpetra/core/src/Tpetra_Experimental_BlockCrsMatrix_def.hpp

```diff
@@ -3066,23 +3066,23 @@ class GetLocalDiagCopy {
             if (debug) {
               const size_t offsetDiff = offset(i+1) - offset(i);
               if (numBytes != offsetDiff) {
-                std::ostream& err = this->markLocalErrorAndGetStream ();  
-                err << prefix 
-                    << "numBytes computed from packRowForBlockCrs is different from "
-                    << "precomputed offset values, LID = " << i << std::endl;
+                std::ostringstream os;
```

Comment:
Wait, don't you want to remember that there was an error, by setting some atomic variable or reducing?  Printing to `cerr` without setting a flag isn't so useful.  See `errorDuringUnpack() = true;` in the loops below for an example.

Reply:
Yes... I wanted to do so but skip it. As now we compute offset values in the same way see line 2961 and 2998 scan, this error checking is probably not much meaningful (I somehow leave it there). Probably I need to remove the checking. If we don't see the errors in PR testing, there is no real chance to encounter runtime error from this logic.

Reply:
@kyungjoo-kim wrote:
> If we don't see the errors in PR testing, there is no real chance to encounter runtime error from this logic.

I just recently found some debug checks like this useful for one of the Sierra apps.  It's a lot harder to write unit tests for user error detection, than to test correct behavior.  While it's not necessary, it does help with diagnosing run-time errors, especially at scale, or where you might not be able to see the whole application or run a debugger.